### ISSUE TITLE
Add input parameter for large payloads on S3

### ISSUE THREAD

**Issue by alexcasalboni** [alexcasalboni/aws-lambda-power-tuning#106]
The maximum input/result size allowed by Step Functions is ~~[32KB](https://docs.aws.amazon.com/step-functions/latest/dg/limits.html#service-limits-task-executions)~~ [256KB](https://aws.amazon.com/it/blogs/compute/introducing-larger-state-payloads-for-aws-step-functions/), which should be enough for most use cases.

But sometimes you need to provide a very large payload to the Lambda function you're power-tuning, and you might even want to provide multiple weighted payloads. In those cases, it'd be useful to provide an S3 object reference (`s3://bucket/key`) and let the `Executor` step fetch it at runtime.

The new input parameter could look as follows:

```json
{
    "lambdaARN": "your-lambda-function-arn",
    "powerValues": [128, 256, 512, 1024],
    "num": 50,
    "payloadS3": "s3://bucket/object.json"
}
```

The implementation should be pretty straightforward. On the other hand, the IAM permission of the `Executor` function should allow it to read an S3 object. We don't want to set another `"*"` in the IAM role (for least-privilege), although it'd be quite inconvenient to setup the S3 path at deploy time.

An alternative solution could be to set up only the S3 Bucket name at deployment time (optionally), and add the corresponding IAM permission only if the CFN Parameter is provided.

**Comment by alexcasalboni** [comment]
WIP :) 

I thought that `payloadS3` would be easier to understand than `payloadReference`.

Feedback is welcome if you think otherwise 🙏

**Comment by alexcasalboni** [comment]
Good news 🚀  The new maximum size is now 256KB (8x larger than before).

**Comment by alexcasalboni** [comment]
Update: this has been on pause for a while since the max payload size update made it less problematic, and I'm still thinking of the best way to handle least-privilege permissions in a backward (and forward) compatible way.

**Comment by Zainabalhaidary** [comment]
Please do add this feature

**Comment by alexcasalboni** [comment]
Update: this is now merged and available on SAR with version `4.2.0`.
